src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoChildFactoryBase.cs

```diff
@@ -73,22 +82,40 @@ protected virtual void OnExpandPopulateNonFolders(IList<TreeNode> allChildren, T
             }
 
             IEnumerable<SmoQuerier> queriers = context.ServiceProvider.GetServices<SmoQuerier>(q => IsCompatibleQuerier(q));
-            var filters = this.Filters;
+            var filters = this.Filters.ToList();
+            if (!string.IsNullOrEmpty(name))
+            {
+                filters.Add(new NodeFilter
+                {
+                    Property = "Name",
+                    Type = typeof(string),
+                    Values = new List<object> { name },
+                });
+            }
             foreach (var querier in queriers)
             {
                 string propertyFilter = GetProperyFilter(filters, querier.GetType(), validForFlag);
-
-                foreach (var smoObject in querier.Query(context, propertyFilter, refresh))
+                try
                 {
-                    if (smoObject == null)
-                    {
-                        Console.WriteLine("smoObject should not be null");
-                    }
-                    TreeNode childNode = CreateChild(parent, smoObject);
-                    if (childNode != null && PassesFinalFilters(childNode, smoObject) && !ShouldFilterNode(childNode, validForFlag))
+                    foreach (var smoObject in querier.Query(context, propertyFilter, refresh))
                     {
-                        allChildren.Add(childNode);
+                        if (smoObject == null)
+                        {
+                            Logger.Write(LogLevel.Error, "smoObject should not be null");
+                        }
+                        TreeNode childNode = CreateChild(parent, smoObject);
+                        if (childNode != null && PassesFinalFilters(childNode, smoObject) && !ShouldFilterNode(childNode, validForFlag))
+                        {
+                            allChildren.Add(childNode);
+                        }
                     }
+
+                }
+                catch (Exception ex)
+                {
+                    string error = string.Format(CultureInfo.InvariantCulture, "Failed getting smo objects. parent:{0} querier: {1} error:{2} inner:{3} stacktrace:{4}",

```

Comment:
Again, long term we need handling of this error handling and somehow return / log in a user-visible way so they can understand failures.